src/pages/config/config-details.js

```diff
@@ -84,125 +86,91 @@ export function ConfigDetailsPage() {
-          {!isLoading ? (
-            Object.entries(jsonLines).map(([lineIndex, line]) => (
-              <div
-                key={lineIndex}
-                style={{
-                  background: checked[lineIndex] ? "#cfe3ff" : "",
-                  borderColor: checked[lineIndex] ? "#cfe3ff" : ""
-                }}
-                className="flex flex-row"
-              >
-                <button
-                  type="button"
-                  onClick={() => handleClick(lineIndex)}
-                  className="flex text-xs mr-2 select-none border-r w-12 justify-between pb-px"
-                >
-                  <span
-                    className="w-4 flex items-center justify-center"
-                    style={{
-                      color: checked[lineIndex] ? "#dd0707" : "#086008"
-                    }}
-                  >
-                    {checked[lineIndex] ? "-" : "+"}
-                  </span>
-                  <div
-                    className="text-xs flex items-center justify-end px-1 text-gray-600"
-                    style={{
-                      borderColor: checked[lineIndex] ? "#cfe3ff" : ""
-                    }}
-                  >
-                    {lineIndex}
-                  </div>
-                </button>
+  const code = useMemo(
+    () =>
+      configDetails?.config && JSON.stringify(configDetails.config, null, 2),
+    [configDetails]
+  );
 
-                <code className="whitespace-pre-wrap text-xs text-gray-800">
-                  {line}
-                </code>
-              </div>
-            ))
-          ) : (
-            <div className="h-32 flex items-center justify-center">
-              <Loading />
-            </div>
-          )}
+  // TODO(ciju): make this lazy. Only needed for IncidentCreate.
+  const configLines = useMemo(() => code && code.split("\n"), [code]);
+
+  const selectedCount = Object.keys(checked).length;
+
+  const selectionControls =
+    selectedCount > 0 ? (
+      <div className="flex flex-row space-x-2">
+        <div className="flex items-center mx-4">
+          {selectedCount} lines selected
         </div>
-        <Modal
-          open={showIncidentModal}
-          onClose={() => setShowIncidentModal(false)}
-          size="small"
-          title="Create New Incident from Selected Evidence"
-        >
-          <IncidentCreate
-            callback={(response) => {
-              navigate(`/incidents/${response.id}`, { replace: true });
-            }}
-            evidence={{
-              configId: id,
-              configName: configDetails?.name,
-              config: jsonLines,
-              type: "config",
-              lines: Object.keys(checked).reduce((acc, lineNum) => {
-                acc[lineNum] = jsonLines[lineNum];
-                return acc;
-              }, {})
-            }}
+        <Button
+          className="btn-secondary"
+          text="Clear"
+          onClick={() => {
```

Comment:
Please refactor all inline handler function expressions with the help of useCallback.

Reply:
[kentcdodds.com/blog/usememo-and-usecallback](https://kentcdodds.com/blog/usememo-and-usecallback) Let's optimize when we see a need for it? Right now, I don't think the interactions has performance bottleneck.